lib/stronger_parameters/constraints/hex_constraint.rb

```diff
@@ -4,7 +4,7 @@
 module StrongerParameters
   class HexConstraint < Constraint
     def value(v)
-      return v if v =~ /\A[a-f0-9]+\z/i
+      return v if v.is_a?(String) && v =~ /\A[a-f0-9]+\z/i
```

Comment:
Would this work?
```suggestion
      return v if v.to_s =~ /\A[a-f0-9]+\z/i
```
Also, why don’t we use `#match?` here?

Reply:
> Also, why don’t we use #match? here?

I presume that’s because non-Strings quietly returned `nil` (until Ruby 3.2), and that was fine.

The original issue: https://github.com/zendesk/stronger_parameters/issues/37. The goal was to handle tokens that were hexadecimal strings, and the original implementation never changed.

Reply:
And since there is this test: [`rejects 123`](https://github.com/zendesk/stronger_parameters/blob/a770d4e2afef658d2e90dcc3f1b35251e1ca3713/test/stronger_parameters/constraints/hex_constraint_test.rb#L14), I think we should not accept integers.